### PR TITLE
New support retrieving container IP from Rancher labels

### PR DESCRIPTION
**Pull request by garceri**
This is a replacement for https://github.com/lyft/metadataproxy/pull/35 which passes the tests and implements the improvements listed on it

### CODE REVIEW

File: metadataproxy/roles.py [hunk 1]
```diff
@@ -138,6 +138,23 @@ def find_container(ip):
...
                     return c
+        # Not Found ? Let's see if we are running under rancher 1.2+,which uses a label to store the IP
+        try:
+            _labels = c.get('Config', {}).get('Labels', {})
+        except (KeyError, ValueError):
+            _labels = {}
+        try:
+            if _labels.get('io.rancher.container.ip'):
+                ip = _labels.get('io.rancher.container.ip').split("/")[0]
+                print(ip)
```
Comment (ryan-lane) [inline metadataproxy/roles.py]: Debugging code? :)

**Comment by ryan-lane** [comment]
Code looks good, after the debugging print statement is removed.

**Comment by garceri** [comment]
Should be good to go @ryan-lane .

**Comment by ryan-lane** [comment]
Thanks for the contribution!

**Comment by ryan-lane** [comment]
released in tag 1.5.0

**Pull request merged**